src/Microsoft.ML.FastTree/FastTreeArguments.cs

```diff
@@ -333,72 +447,130 @@ public abstract class BoostedTreeArgs : TreeArgs
+        /// </summary>
         [Argument(ArgumentType.LastOccurenceWins, HelpText = "Optimization algorithm to be used (GradientDescent, AcceleratedGradientDescent)", ShortName = "oa")]
         public OptimizationAlgorithmType OptimizationAlgorithm = OptimizationAlgorithmType.GradientDescent;
 
+        /// <summary>
+        /// Early stopping rule. (Validation set (/valid) is required).
+        /// </summary>
         [Argument(ArgumentType.Multiple, HelpText = "Early stopping rule. (Validation set (/valid) is required.)", ShortName = "esr", NullName = "<Disable>")]
         [TGUI(Label = "Early Stopping Rule", Description = "Early stopping rule. (Validation set (/valid) is required.)")]
         public IEarlyStoppingCriterionFactory EarlyStoppingRule;
 
+        /// <summary>
+        /// Early stopping metrics. (For regression, 1: L1, 2:L2; for ranking, 1:NDCG@1, 3:NDCG@3).
+        /// </summary>
         [Argument(ArgumentType.AtMostOnce, HelpText = "Early stopping metrics. (For regression, 1: L1, 2:L2; for ranking, 1:NDCG@1, 3:NDCG@3)", ShortName = "esmt")]
         [TGUI(Description = "Early stopping metrics. (For regression, 1: L1, 2:L2; for ranking, 1:NDCG@1, 3:NDCG@3)")]
         public int EarlyStoppingMetrics;
 
+        /// <summary>
+        /// Enable post-training pruning to avoid overfitting. (a validation set is required).
+        /// </summary>
         [Argument(ArgumentType.AtMostOnce, HelpText = "Enable post-training pruning to avoid overfitting. (a validation set is required)", ShortName = "pruning")]
         public bool EnablePruning;
 
+        /// <summary>
+        /// Use window and tolerance for pruning.
+        /// </summary>
         [Argument(ArgumentType.LastOccurenceWins, HelpText = "Use window and tolerance for pruning", ShortName = "prtol")]
         public bool UseTolerantPruning;
 
+        /// <summary>
+        /// The tolerance threshold for pruning.
+        /// </summary>
         [Argument(ArgumentType.AtMostOnce, HelpText = "The tolerance threshold for pruning", ShortName = "prth")]
         [TGUI(Description = "Pruning threshold")]
         public Double PruningThreshold = 0.004;
 
+        /// <summary>
+        /// The moving window size for pruning.
+        /// </summary>
         [Argument(ArgumentType.AtMostOnce, HelpText = "The moving window size for pruning", ShortName = "prws")]
         [TGUI(Description = "Pruning window size")]
         public int PruningWindowSize = 5;
 
+        /// <summary>
+        /// The learning rate.
+        /// </summary>
         [Argument(ArgumentType.LastOccurenceWins, HelpText = "The learning rate", ShortName = "lr", SortOrder = 4)]
         [TGUI(Label = "Learning Rate", SuggestedSweeps = "0.025-0.4;log")]
         [TlcModule.SweepableFloatParamAttribute("LearningRates", 0.025f, 0.4f, isLogScale: true)]
         public Double LearningRates = Defaults.LearningRates;
 
+        /// <summary>
+        /// Shrinkage.
+        /// </summary>
```

Comment:
Would be great to have more than just a single word to describe this. Maybe on the tlc documentation website there is more information on this parameter.

Reply:
i guess its one of those known terms in Gradient Boosted trees. do u have any suggestions ?

---
In reply to: [245964690](https://github.com/dotnet/machinelearning/pull/2047#discussion_r245964690) [](ancestors = 245964690)

Reply:
Maybe "The shrinkage". But yes I looked it up and you are right, so not necessary.

---
In reply to: [246105587](https://github.com/dotnet/machinelearning/pull/2047#discussion_r246105587) [](ancestors = 246105587,245964690)